### PR TITLE
Adding loader for TAU NIGENS SSE 2021 dataset

### PR DESCRIPTION
**Pull request by iranroman**
### Description

Please include the following information at the top level docstring for the dataset's module mydataset.py:

- [x] Describe annotations included in the dataset
- [x] Indicate the total duration of the dataset in hours, and (optionally) also list the number of individual files
- [x] Mention the origin of the dataset (e.g. creator, institution)
- [x] Describe the type of audio included in the dataset
- [x] Indicate any relevant papers related to the dataset
- [x] Include a description about how the data can be accessed and the license it uses (if applicable)

#### Dataset loaders checklist:

- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`, which generates an index file.
- [x] Run the script on the canonical version of the dataset and save the index in `soundata/indexes/` e.g. `my_dataset_index.json`.
- [x] Create a module in soundata, e.g. `soundata/my_dataset.py`
- [x] Create tests for your loader in `tests/`, e.g. `test_my_dataset.py`
- [x] Add your module to `docs/source/soundata.rst` and `docs/source/quick_reference.rst`
- [x] Run `tests/test_full_dataset.py` on your dataset.

### CODE REVIEW

File: docs/source/tutorial.rst [hunk 1]
```diff
@@ -426,7 +426,7 @@ The following is a simple example of a generator that can be used to create a te
                 }
 
         dataset = tf.data.Dataset.from_generator(
-            urbansound8k_generator('urbansound8k'),
+            data_generator('urbansound8k'),
```
Comment (magdalenafuentes) [inline docs/source/tutorial.rst:429]: :heart:

File: soundata/datasets/tau_nigens_sse_2021.py [hunk 2]
```diff
@@ -0,0 +1,652 @@
...
+    :class: dropdown
+
+    *TAU NIGENS Spatial Sound Events: scene recordings with (moving) sound events of distinct categories*
+
+    *Created By:*
+        Archontis Politis, Sharath Adavanne, Tuomas Virtanen.
+        Audio Research Group, Tampere University (Finland). Version 1.2.0
+
+    *Description:*
+        DESCRIPTION:
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_nigens_sse_2021.py]: description appears twice

File: soundata/datasets/tau_nigens_sse_2021.py [hunk 3]
```diff
@@ -0,0 +1,652 @@
...
+        * Up to three overlapping sound events possible, temporally and spatially.
+        * Simultaneous directional interfering sound events with their own temporal activities, static or moving.
+        * Realistic spatial ambient noise collected from each room is added to the spatialized sound events, at varying signal-to-noise ratios (SNR) ranging from noiseless (30dB) to noisy (6dB) conditions.
+
+    *Annotations Included:*
+        * Each recording in the development set has labels of events and DoAs in a plain csv file with the same filename.
+        * Each row in the csv file has a frame number, active class index, event number index, azimuth, and elevation.
+        * Frame, class, and track enumeration begins at 0. 
+        * Frames correspond to a temporal resolution of 100msec. 
+        * Azimuth and elevation angles are given in degrees, rounded to the closest integer value, with azimuth and elevation being zero at the front, azimuth $\phi \in [-180^{\circ}, 180^{\circ}]$, and elevation $\theta \in [-90^{\circ}, 90^{\circ}]$. Note that the azimuth angle is increasing counter-clockwise ($\phi = 90^{\circ}$ at the left).
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_nigens_sse_2021.py]: Same comment as the other loader, the equations don't render properly in the docs ([see here](https://soundata--77.org.readthedocs.build/en/77/source/soundata.html#tau-nigens-sse-2021))

File: soundata/datasets/tau_nigens_sse_2021.py [hunk 4]
```diff
@@ -0,0 +1,652 @@
...
+        * The development dataset is split in training, validation, and test sets.
+        * The training set consists of 400 recordings.
+        * The validation set consists of 100 recordings.
+        * The test set consists of 100 recordings.
+        * The evalutation dataset constists of 200 recordings.
+
+    *Please Acknowledge TAU-NIGENS SSE 2021 in Academic Research:*
+        * If you use this dataset please cite the report on its creation, and the corresponding DCASE2021 task setup:
+
+            * Archontis Politis, Sharath Adavanne, Daniel Krause, Antoine Deleforge, Prerak Srivastava, and Tuomas Virtanen. A dataset of dynamic reverberant sound scenes with directional interferers for sound event localization and detection. arXiv preprint arXiv:2106.06999, 2021. URL: https://arxiv.org/abs/2106.06999, arXiv:2106.06999.
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_nigens_sse_2021.py:57]: Do they have a published version?

File: soundata/datasets/tau_nigens_sse_2021.py [hunk 5]
```diff
@@ -0,0 +1,652 @@
...
+import json
+import glob
+import numbers
+from itertools import cycle
+
+from soundata import download_utils
+from soundata import jams_utils
+from soundata import core
+from soundata import annotations
+from soundata import io
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_nigens_sse_2021.py]: One line would be better!

File: soundata/datasets/tau_nigens_sse_2021.py [hunk 6]
```diff
@@ -0,0 +1,652 @@
...
+BIBTEX = """
+@article{politis2021dataset,
+    author = "Politis, Archontis and Adavanne, Sharath and Krause, Daniel and Deleforge, Antoine and Srivastava, Prerak and Virtanen, Tuomas",
+    title = "A Dataset of Dynamic Reverberant Sound Scenes with Directional Interferers for Sound Event Localization and Detection",
+    year = "2021",
+    journal = "arXiv preprint arXiv:2106.06999",
+    eprint = "2106.06999",
+    archiveprefix = "arXiv",
+    primaryclass = "eess.AS",
+    url = "https://arxiv.org/abs/2106.06999"
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_nigens_sse_2021.py:86]: published anywhere?

File: tests/datasets/test_tau_nigens_sse_2021.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
+import os
+import numpy as np
+import pytest
+
+from tests.test_utils import run_clip_tests
+
+from soundata import annotations
+from soundata.datasets import tau_nigens_sse_2021
+from tests.test_utils import DEFAULT_DATA_HOME
```
Comment (magdalenafuentes) [inline tests/datasets/test_tau_nigens_sse_2021.py]: Maybe add this to line 5?

**Review by magdalenafuentes — APPROVED** [review]
@iranroman just left a couple minor comments and good to merge!

File: soundata/datasets/tau_nigens_sse_2021.py (hunk 3 above)
Comment (iranroman) [reply soundata/datasets/tau_nigens_sse_2021.py]: Do we know if we support LaTex at all?

File: soundata/datasets/tau_nigens_sse_2021.py (hunk 4 above)
Comment (iranroman) [reply soundata/datasets/tau_nigens_sse_2021.py:57]: Not yet. Maybe after the DCASE workshop

File: soundata/datasets/tau_nigens_sse_2021.py (hunk 6 above)
Comment (iranroman) [reply soundata/datasets/tau_nigens_sse_2021.py:86]: not yet

**Pull request merged**